llvm-external-projects/iree-dialects/test/iree_linalgext/tiling.mlir

```diff
@@ -1206,3 +1206,43 @@ func @pad_tensor(%arg0 : tensor<?x?xf32>, %arg1 : index, %arg2 : index,
 //      CHECK:       scf.yield %[[INSERT]]
 //      CHECK:     scf.yield %[[YIELD]]
 //      CHECK:   return %[[RESULT]]
+
+// -----
+
+func @scan_1d(%0: tensor<128xi32>) -> tensor<128xi32> {
```

Comment:
This didnt actually tile the operation did it? You need to add `{__internal_linalg_transform__ = "tiling_input"}` to the op (its an extra attribute to tell the pass to tile the operation).

Reply:
Thanks! Yes, I missed that. When I do that, and modify the IR to be
```mlir
  func @scan_1d(%0: tensor<128xi32>) -> tensor<128xi32> {
    %c0 = arith.constant 0 : i32
    %1 = linalg.init_tensor [128] : tensor<128xi32>
    %2 = iree_linalg_ext.scan
      dimension(0) inclusive(true)
      {__internal_linalg_transform__ = "tiling_input"}
      identity(%c0 : i32)
      ins(%0 : tensor<128xi32>) outs(%1 : tensor<128xi32>) {
      ^bb0(%arg0 : i32, %arg1 : i32):
        %sum = arith.addi %arg0, %arg1 : i32
        iree_linalg_ext.yield %sum : i32
    } -> tensor<128xi32>
    return %2 : tensor<128xi32>
  }
```

Then, I get the following error
```
error: 'iree_linalg_ext.scan' op unimplemented tiling of non-parallel loop iterator type
```

But I thought I provided a getTiledImplementation, so not sure why I would be hitting this error. Any ideas on what could be causing this?

Thanks!

Reply:
Instead of creating a new set of tiling options, I just re-used "outer_reduce_input". Let me know what you think.

Reply:
Sorry, I missed this. What you did is right. You cannot specify a tiling value for a reduced iterator. Thats the error. Its just a matter of specifying the correct configuration for tiling. When plugged into IREE, we will have to specify a way to handle this if needed, but for testing what you have is correct.